src/iotjs_binding.h

```diff
@@ -251,36 +201,41 @@ iotjs_jval_t iotjs_jval_create_function_with_dispatch(
+    JNativeInfoType* out_native_info;                                         \
+    jerry_get_object_native_pointer(jthis, (void**)&name, &out_native_info);  \
+    if (!name || out_native_info != &this_module_native_info) {               \
+      return jerry_create_error(JERRY_ERROR_COMMON, (const jerry_char_t*)""); \
+    }                                                                         \
+  } while (0)
 
-#define DJHANDLER_GET_REQUIRED_CONF_VALUE(src, target, property, type)        \
+#define JHANDLER_DECLARE_OBJECT_PTR(index, type, name)                        \
+  iotjs_##type##_t* name;                                                     \
   do {                                                                        \
-    iotjs_jval_t jtmp = iotjs_jval_get_property(src, property);               \
-    if (jerry_value_is_undefined(jtmp)) {                                     \
-      JHANDLER_THROW(TYPE, "Missing argument, required " property);           \
-      return;                                                                 \
-    } else if (jerry_value_is_##type(jtmp))                                   \
-      target = iotjs_jval_as_##type(jtmp);                                    \
-    else {                                                                    \
-      JHANDLER_THROW(TYPE,                                                    \
-                     "Bad arguments, required " property " is not a " #type); \
-      return;                                                                 \
+    JNativeInfoType* out_native_info;                                         \
+    jerry_get_object_native_pointer(jargv[index], (void**)&name,              \
+                                    &out_native_info);                        \
+    if (!name || out_native_info != &this_module_native_info) {               \
+      return jerry_create_error(JERRY_ERROR_COMMON, (const jerry_char_t*)""); \
     }                                                                         \
-    jerry_release_value(jtmp);                                                \
-  } while (0);
+  } while (0)
+
+#define DJHANDLER_GET_REQUIRED_CONF_VALUE(src, target, property, type)       \
+  do {                                                                       \
+    iotjs_jval_t jtmp = iotjs_jval_get_property(src, property);              \
+    if (iotjs_jval_is_undefined(jtmp)) {                                     \
+      return jerry_create_error_sz(TYPE,                                     \
+                                   "Missing argument, required " property);  \
+    } else if (iotjs_jval_is_##type(jtmp))                                   \
+      target = iotjs_jval_as_##type(jtmp);                                   \
+    else {                                                                   \
+      return jerry_create_error_sz(TYPE, "Bad arguments, required " property \
```

Comment:
I'm not sure, but shouldn't the `jtmp` be released before returning?

Reply:
Right, I'll fix it.